widgetastic_manageiq/__init__.py

```diff
@@ -1097,6 +1099,138 @@ def workaround_save_issue(self):
+    """
+
+    @ParametrizedView.nested
+    class list(ParametrizedView):  # noqa
+        PARAMETERS = ("list_name", "item_name",)
+        list_item = Text(ParametrizedLocator('//div[@id={list_name|quote}]/../div'
+            '//a[normalize-space(.)={item_name|quote}]'))
+
+        def list_click(self):
+            """Clicks the list item with this name."""
+
+            return self.list_item.click()
+
+    def __init__(self, parent, list_name, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.list_name = list_name
+
+    def click_at(self, item_name):
+        """Clicks the list item with this name.
+
+        Args:
+            item_name: Name of the item
+        """
+        return self.list(self.list_name, item_name).list_click()
+
+
+class SSUIDropdown(Dropdown):
+    """Represents the SSUI dropdown."""
+
+    ROOT = ParametrizedLocator(
+        '//button[contains(@class, "dropdown-toggle")]/span[normalize-space(.)="Configuration"]')
+    BUTTON_LOCATOR = '//button'
+    ITEMS_LOCATOR = './ul/li/a'
+    ITEM_LOCATOR = '//ul/li/a[normalize-space(.)={}]'
+
+    def __init__(self, parent, text, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.text = text
+
+
+class SSUIPrimarycard(Widget, ClickableMixin):
+    """Represents a primary card on dashboard like Total Service or Total Requests."""
+
+    @ParametrizedView.nested
+    class primary_card(ParametrizedView):  # noqa
+        PARAMETERS = ("item_name",)
+        card = Text(ParametrizedLocator('//div[@class="ss-dashboard__card-primary__count"]'
+            '/h3[normalize-space(.)={item_name|quote}]'))
+        count = Text(ParametrizedLocator('//div[@class="ss-dashboard__card-primary__count"]/h2'
+            '[./following-sibling::h3[normalize-space(.)={item_name|quote}]]'))
+
+        def card_click(self):
+            """Clicks the primary card with this name."""
+
+            return self.card.click()
+
+        def card_count(self):
+            """Gets the count displayed on card"""
+
+            return self.browser.text(self.count)
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def click_at(self, item_name):
+        """Clicks the card item with this name.
+
+        Args:
+            item_name: Name of the card
+        """
+        return self.primary_card(item_name).card_click()
+
+    def get_count(self, item_name):
+        """
+        Returns the count displayed on UI.
+
+        Args:
+            item_name: Name of the card
+        """
+        return self.primary_card(item_name).card_count()
+
+
+class SSUIAggregatecard(Widget, ClickableMixin):
+    """Represents an aggregate card like Current Services or Retired services."""
+
+    @ParametrizedView.nested
+    class aggregate_card(ParametrizedView):  # noqa
+        PARAMETERS = ("item_name",)
+        card = Text(ParametrizedLocator('//div[@class="card-pf-body"]'
```

Comment:
Hmm, with parametrized locators not starting with . you are risking matching multiple things.